clang/lib/AST/ByteCode/InterpState.cpp

```diff
@@ -27,7 +27,7 @@ bool InterpState::inConstantContext() const {
 }
 
 InterpState::~InterpState() {
-  while (Current) {
+  while (Current && !Current->isBottomFrame()) {
```

Comment:
This is causing use-after-destruction errors for me when running clang tests under MemorySanitizer.

If I understand correctly, after the body of `~EvalEmitter()` finishes running, its members are destroyed in reverse order of declaration, and `BottomFrame` gets destroyed before the destructor for `InterpState S;` runs.

Reply:
Some reordering is needed for "InterpState S;" and "InterpFrame BottomFrame;".